.github/workflows/build_wheels.yml

```diff
@@ -24,32 +24,33 @@ jobs:
       - name: Build sdist
         run: pipx run build --sdist
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
+          name: sdist
           path: dist/*.tar.gz
 
-  choose_architectures:
-    name: Decide which architectures to build wheels for
+  choose_wheel_types:
+    name: Decide which wheel types to build
     runs-on: ubuntu-latest
     steps:
-      - id: x86_64
-        run: echo "cibw_arch=x86_64" >> $GITHUB_OUTPUT
-      - id: aarch64
+      - id: manylinux_x86_64
+        run: echo "wheel_types=manylinux_x86_64" >> $GITHUB_OUTPUT
+      - id: musllinux_x86_64
+        run: echo "wheel_types=musllinux_x86_64" >> $GITHUB_OUTPUT
+      - id: manylinux_aarch64
         if: github.event_name == 'release' && github.event.action == 'published'
-        run: echo "cibw_arch=aarch64" >> $GITHUB_OUTPUT
+        run: echo "wheel_types=manylinux_aarch64" >> $GITHUB_OUTPUT
     outputs:
-      cibw_arches: ${{ toJSON(steps.*.outputs.cibw_arch) }}
+      wheel_types: ${{ toJSON(steps.*.outputs.wheel_types) }}
 
   build_wheels:
-    needs: [build_sdist, choose_architectures]
-    name: Wheel for Linux-${{ matrix.cibw_python }}-${{ matrix.cibw_arch }}
-    runs-on: ${{ matrix.os }}
+    needs: [build_sdist, choose_wheel_types]
+    name: ${{ matrix.wheel_type }} wheels
```

Comment:
I would add `Build ` at the front there to make it clearer what this step does.

```suggestion
    name: Build ${{ matrix.wheel_type }} wheels
```

Reply:
That gets truncated in the UI:

![image](https://github.com/bloomberg/pystack/assets/29423/838812fb-5a1e-43c5-975f-ba6d5631af49)

Compared to without:

![image](https://github.com/bloomberg/pystack/assets/29423/7af8c21e-83b0-4935-a59a-1939c1943b12)

I think it looks better without the word "build", and I think the intention is clear enough given that it's in a box labeled "Matrix: build_wheels", so I think I'm gonna skip this suggestion.

Reply:
I guess it depends on what you prioritize, between the UI in the PR checks (where adding `Build` probably makes more sense) or the GH actions summary view (where it's better left off). Either way, this was just a nitpick, it is indeed easy enough to understand.